src/stores/video.ts

```diff
@@ -57,13 +57,95 @@ export const useVideoStore = defineStore('video', () => {
   const timeNow = useTimestamp({ interval: 500 })
   const autoProcessVideos = useBlueOsStorage('cockpit-auto-process-videos', true)
   const lastRenamedStreamName = ref('')
+  const streamsPreparingToRecord = ref<string[]>([])
 
   const namesAvailableStreams = computed(() => mainWebRTCManager.availableStreams.value.map((stream) => stream.name))
 
+  const getRecorderOptions = async (streamData: StreamData): Promise<MediaRecorderOptions> => {
+    // Check on WebRTCManager to exposed RTCPeerConnection details.
+    const waitForPeerConnection = async (): Promise<RTCPeerConnection | null> => {
+      const until = Date.now() + 1_500
+      while (Date.now() < until) {
+        const pc =
+          (streamData.webRtcManager as any).peerConnection ?? (streamData.webRtcManager as any).session?.peerConnection
+        if (pc) return pc as RTCPeerConnection
+        await new Promise((r) => setTimeout(r, 80))
+      }
+      return null
+    }
+
+    const peerConnection = await waitForPeerConnection()
+    let detectedMime: string | null = null
+
+    if (peerConnection?.getStats) {
+      const stats = await peerConnection.getStats()
+      stats.forEach((r) => {
+        if (r.type === 'inbound-rtp' && r.kind === 'video' && r.codecId) {
+          const codec = stats.get(r.codecId)
+          if (codec?.mimeType) detectedMime = codec.mimeType.toLowerCase()
+        }
+      })
+      if (detectedMime) console.debug('[Recorder] Detected codec via getStats →', detectedMime)
+    }
+    // SDP fallback – if getStats doesn't expose the coded yet, parse it out of the SDP
+
+    if (!detectedMime && peerConnection?.remoteDescription) {
+      const sdp = peerConnection.remoteDescription.sdp
+      const match = sdp.match(/^m=video .+$/m)
+      if (match) {
+        const pts = match[0].split(' ').slice(3)
+        for (const pt of pts) {
+          const rp = new RegExp(`a=rtpmap:${pt} ([A-Za-z0-9]+)/`).exec(sdp)
+          if (rp) {
+            detectedMime = `video/${rp[1].toLowerCase()}`
+            console.debug('[Recorder] Detected codec via SDP →', detectedMime)
+            break
+          }
+        }
+      }
+    }
+
+    if (detectedMime) {
+      if (MediaRecorder.isTypeSupported(detectedMime)) {
+        console.debug('[Recorder] Using detected codec.')
+        return { mimeType: detectedMime }
+      }
+
+      const codecOnly = detectedMime.replace(/^video\//, '')
```

Comment:
There are a lot of cryptic strings here. Can we puts some comments here and there explaining exactly what is happening, so people reading this code in the future can better situate themselves?